job/server/src/main/java/alluxio/worker/job/task/TaskExecutorManager.java

```diff
@@ -61,11 +64,43 @@ public TaskExecutorManager(int taskExecutorPoolSize, WorkerNetAddress address) {
     mTaskFutures = Maps.newHashMap();
     mUnfinishedTasks = Maps.newHashMap();
     mTaskUpdates = Maps.newHashMap();
-    mTaskExecutionService = Executors.newFixedThreadPool(taskExecutorPoolSize,
-      ThreadFactoryUtils.build("task-execution-service-%d", true));
+    mTaskExecutionService = new PausableThreadPoolExecutor(taskExecutorPoolSize,
+        MAX_TASK_EXECUTOR_POOL_SIZE, 0L, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>(),
+        ThreadFactoryUtils.build("task-execution-service-%d", true));
     mAddress = address;
   }
 
+  /**
+   * @return number of active tasks
+   */
+  public int getNumActiveTasks() {
+    return mTaskExecutionService.getNumActiveTasks();
+  }
+
+  /**
+   * @return task executor pool size
+   */
+  public int getTaskExecutorPoolSize() {
+    return mTaskExecutionService.getCorePoolSize();
```

Comment:
This will always return `taskExecutorPoolSize` (# of core threads), is that what we want?

Reply:
?? Doesn't that make sense it's getTaskExecutorPoolSize.